lib/fileset/README.md

```diff
@@ -138,6 +138,33 @@ Arguments:
   - (-) Leaves us with no identity element for `union` and no reasonable return value for `unions []`.
     From a set theory perspective, which has a well-known notion of empty sets, this is unintuitive.
 
+### No intersection for lists
+
+While there is `intersection a b`, there is no function `intersections [ a b c ]`.
+
+Arguments:
+- (+) There is no known use case for such a function, it can be added later if a use case arises
+- (+) There is no suitable return value for `intersections [ ]`, see also "Nullary intersections" [here](https://en.wikipedia.org/w/index.php?title=List_of_set_identities_and_relations&oldid=1177174035#Definitions)
+  - (-) Could throw an error for that case
+  - (-) Create a special value to represent "all the files" and return that
+    - (+) Such a value could then not be used with `fileFilter` unless the internal representation is changed considerably
+  - (-) Could return the empty file set
+    - (+) This would be wrong in set theory
+- (-) Inconsistent with `union` and `unions`
```

Comment:
@roberth @fricklerhandwerk This PR isn't done yet, but I wonder what you think of this design decision.

It's an interesting duality:
- Union is mostly needed between _many_ file sets (e.g. combining together the exact files you need)
- Intersection is mostly needed between _few_ file sets (e.g. limiting the selected files to some directory)

We might really only need `unions` (taking a list) and `intersection` (binary operation).
`union` (binary operation) doesn't seem that important, but I can imagine some situations where it could be useful and we have it already.

Notably Haskell's [`Data.Set`](https://hackage.haskell.org/package/containers-0.6.7/docs/Data-Set.html#g:7) also provides `union`, `unions`, `intersection` but no `intersections` (there is one, but it's [in `Data.Set.Internal`](https://hackage.haskell.org/package/containers-0.6.7/docs/Data-Set-Internal.html#v:intersections)).

Reply:
We could have an interface that requires at least one set: `intersect :: a -> [a] -> a`. But probably a binary one is good enough. One can still fold over a list, then defining the base set will be required for sane results. That again we can add to documentation as an example.

Reply:
In math it's easy to invoke a somewhat implicitly defined "universe" set. Us finite mortals would have to make that explicit, which is another way to land on `a -> List a -> a` whereas @fricklerhandwerk seems to have arrived through `NonEmptyList a -> a`.